### DISCUSSION TITLE
Add support for SIBIONICS CGM sensors

### DISCUSSION THREAD

**Discussion by Allmantool** [j-kaltes/Juggluco#100] [Ideas]
HI there !
A new China CGM has received CE mark approval in Europe for its GS1 continuous glucose monitoring (CGM) system.
And it looks like a decent competitor for Freestyle Libre 2

Why is it worth to extend Juggluco with Sibionics CGM support?
- The sensor price & abilities very alike to Libre 2
- Evermore `**Sibionics is able to work up to 24 days (instead of 14 days for Freestly Libre**`). 
Note: 
functionality hasn't been provided by the official vendor by default, but, as I can see, the sensor can be tweaked, without a big effort. 
At least it has been done already once so far (https://t.me/kubarev_modz (24 days mode for Android))

For more details, pls. take a look at links below

- https://www.sibionicscgm.com/
- https://www.drugdeliverybusiness.com/sibionics-ce-mark-14-day-cgm/
- https://github.com/NightscoutFoundation/xDrip/discussions/3063
- https://t.me/kubarev_modz (24 days mode for Android)
- http://en.sibionics.com/Consumer/index.html

**Comment by j-kaltes** [comment]
Yes, it would be nice when Juggluco would be available to users of Dexcom and Symbionese sensors.

**Reply by AgushkaNightscout** [reply]
If it would be acceptable to you, I can send you a couple of Sibionics sensors by post for experiments.

**Reply by j-kaltes** [reply]
Yes, if I receive a few sensors, I would try to make them work with Juggluco. 
I don’t think that it would be more difficult, than connecting Libre 3 sensors.
Do you use Sibionics sensors?

**Reply by AgushkaNightscout** [reply]
Yes, I used 2 sensors, but I was not satisfied with the stability of the connection - constant interruptions. Users defeat it with Tasker, but it's an obvious kludge. Stability of readings, size, accuracy satisfied me quite well. Y.Kubarev's app sends data to Хdrip+ (thus, calibration is possible) and allows you to use the sensor for 24 days (probably more, but the internal memory of the sensor is enough for just that many 5-minute readings), which is economical and convenient. The sensor is rumored to be able to read every minute. A good candidate for Juggluco upgrade)

I'll email you for details, ok?

**Reply by j-kaltes** [reply]
You used [SugarJoy](https://t.me/kubarev_modz/400)  ?

You can e-mail to jaapkorthalsaltes@gmail.com

**Reply by AgushkaNightscout** [reply]
Yes, but an earlier version. Judging by the feedback in the Telegram group, the breaks haven't gone anywhere with the latest version. The problem is not even in the actual disconnections, but in the fact that reconnection with the sensor is not automatic, and you have to manually connect it in application (in my case). [Here](https://docs.google.com/spreadsheets/d/1c4rPYr5i4cidGBLOSi5kE4WRqgZGHVXaMdOUwS_TDvQ/edit#gid=1343440576) is a file with user reports on disconnections (да=stable connection, нет=breaks) for 32-bit version app, but it has opposite results even for the same phone model. The problem is in the original Chinese application and Yaroslav wrote about it in his Telegram group.

**Comment by j-kaltes** [comment]
Sibionics sensors can be used with Juggluco 8.0.0.
https://www.juggluco.nl/Juggluco/download.html

**Comment by aptash** [comment]
Hello!
First of all thanks a lot for the app!

Today I tried Juggluco (8.0.7) with Sibionics (sensor age: 19d9h) and had some problems.

Please compare graphs:
Juggluco:
![image](https://github.com/j-kaltes/Juggluco/assets/5793551/ef88dfe6-5eca-4ea9-9aee-84e9cb4baaa7)

SugarJoy+:
![image](https://github.com/j-kaltes/Juggluco/assets/5793551/5ea77253-b61b-4b17-ac0d-1007f4e9ffe7)

The actual BG at the very end of graph was 15,5 mmol/l by glucometer.

Also I noticed that though Juggluco reads data every minute actually all values are the same during 5 minutes:
![image](https://github.com/j-kaltes/Juggluco/assets/5793551/11808597-72d0-46ac-aea3-c2b45b4296f5)

**Reply by j-kaltes** [reply]
This is another version that only generates every 5 minutes a value and has somewhat different values. It probably works best when used from the start of the sensor, but can also be used otherwise.

It is also restricted to only 572 hours (23 days and 20 hour) instead of 24 days, because the sensor I tried didn't last longer than that. It instead gave the same values with different dates when used longer.

EDIT: 
Waits 6 minutes before no longer displaying the current value instead of 5 minutes (previous version) or 3 minutes (all other versions):
PHONE: https://drive.google.com/file/d/1azrDXIqLyy1zRmU4LBDWUCr_sGw9dZUo/view?usp=drive_link
WATCH: https://drive.google.com/file/d/1HpdTiylNKNZYUaTttZW6oWruKR5FPDwg/view?usp=drive_link

**Reply by zachem-to** [reply]
Hello, thank you very much for your work!
Will future updates include the ability to get minute-by-minute readings?

**Reply by aptash** [reply]
Thanks for 5 minutes values!
Unfortunately you said nothing about notable difference in BG values between Sibionics patched app and Juggluco.
Don't you think that Juggluco shows suspiciously straight line?

**Reply by zachem-to** [reply]
Looks like over-smoothing.. Have you tried the newest update 8.0.9 or 8.0.10? I didn't notice anything like that there

**Comment by zachem-to** [comment]
Please help me understand. My xdrip readings are 5 minutes behind my juggluco readings. Is this correct or have I configured something wrong?

**Reply by j-kaltes** [reply]
When you send values with the Patched Libre broadcast xDrip modifies the values. Juggluco 8.0.10 added the EverSense broadcast. Values send with the EverSense broadcast are not modified that way by xDrip.
https://www.juggluco.nl/Juggluco/download.html
See: https://www.juggluco.nl/Jugglucohelp/settinghelp.html#eversense

**Reply by zachem-to** [reply]
Thank you very much! But EverSence requires a calibration plugin, Libre(patched) is good

**Reply by aptash** [reply]
You can use EverSense without calibration.
Calibration is an option.

**Reply by zachem-to** [reply]
I know that, thank you. But sometimes it's required

**Comment by j-kaltes** [comment]
> Thanks for 5 minutes values!
> Unfortunately you said nothing about notable difference in BG values between Sibionics patched app and Juggluco.
> Don't you think that Juggluco shows suspiciously straight line?

I never saw such a straight line. I can look at it what caused it. For that you need to run a newly installed logging version of Juggluco and receive all glucose values anew in that app. You can download it from: https://www.juggluco.nl/Juggluco/download.html#bugreport
You get the log file out of the device with: 
`adb shell run-as tk.glucodata cat /data/data/tk.glucodata/files/logs/trace.log > trace.log`
For this you first need to uninstall Juggluco. 
For if you don't want to do that, here is a link to a version of Juggluco with another application id so that you can install it besides your currently installed version without uninstalling it:
https://drive.google.com/open?id=1TD8La3a8rSfBSHAJBoGB6egoYiKia4Ns
Then you need to use the following command to get the log file out of the device:
`adb shell run-as tk.glucodata.debug cat /data/data/tk.glucodata.debug/files/logs/trace.log > trace.log`

E-Mail trace.log to jaapkorthalsaltes@gmail.com
E-Mail also the exported values (left middle menu->Export->Stream). You can also send the glucose values of SugarJoy+.
Make sure that all the time **ONLY ONE APP IS RUNNING**.

**Reply by aptash** [reply]
Thanks!
We changed the sensor a couple days ago.
I will try some other day then.

**Reply by j-kaltes** [reply]
You can also receive the data from an old sensor. For this use the the app with another application ID. You should get all data the first time without restarting the app. It does receive the same values, but ending in the current time instead of the time the last value was measured. So the times are wrong.

**Reply by aptash** [reply]
Oh! Great! I saved the sensor and can do it by using another phone and the logging version of Juggluco.

**Reply by aptash** [reply]
Hello!
I send you an email. I think that maybe my email is too late and we don’t have the discussed problem already in 8.0.14.
It seems, when I connected the old sensor, I have date and time offset. But I think I found the same timeframe and it looks fine now.
Thanks a lot for your constant updates!

Here is it looks like the same timeframe:
![image](https://github.com/j-kaltes/Juggluco/assets/5793551/831432a0-f0db-41e2-8a4f-b4f607da3774)

**Comment by thejazzs** [comment]
@j-kaltes 
Thank you for your great work.
It would be pleased and appreciated If you may kindly clarify the below questions shortly regarding Sibionics EU version ; 

- Can we use Sibionic companion app and Juggluco together ?
- Can we use Juggluco while we are using the companion app? (This is question for the sensors in use)
- Can we calibrate the Sibionics sensors via Juggluco ? 
- Can we extend Sibionics sensors via Juggluco ? 
- I couldn't find a manual for these kind of informations, If there is, it would be highly appreciated to be shared. 

I will be waiting your kind reply, thank you.

**Reply by j-kaltes** [reply]
- You certainly can't connect the same sensor at the same time to two apps;
- You can switch a sensor you have previously used with Sibionics own app to Juglguco;
- You can't calibrate sensors within Juggluco;
- I expect EU Sibionics GS1 sensors to last up to nearly 24 days, as did the Chinese variant.

**Reply by thejazzs** [reply]
Thank you for the reply.

- You certainly can't connect the same sensor at the same time to two apps; 
    _**CLEAR**_
- You can switch a sensor you have previously used with Sibionics own app to Juglguco;      _**How to do that? By unistalling the Sibionics' own app or else?**_
- You can't calibrate sensors within Juggluco;       _**CLEAR**_
- I expect EU Sibionics GS1 sensors to last up to nearly 24 days, as did the Chinese variant.       _**Can't it count down from 14 days? Or does it any place to arrange it? Or we just let it go? Or else?**_

**Reply by thejazzs** [reply]
@j-kaltes
I started a sibionics sensor with juggluco.
- Juggluco is reading glucose values per minute instead of per 5 minutes. Does it reduce the life of the sensor? 
- I am taking some readings from blood when the readings are stable and sometimes it is higher and sometimes it is lower than juggluco 's readings. When I was using sibionics companion app, mostly blood readings was higher. I was using xdrip's calibration and it was good. Right now I cannot use calibration safely due to blood results are sometimes higher sometimes lower than Juggluco's values. Why do you think it is happening like this? 
- Is there any way to change Juggluco's reading to 5 minutes?

**Reply by j-kaltes** [reply]
• The sensor always sends glucose values every minute. Juggluco asks one time for glucose values from a certain index and the sensor starts sending glucose values in a pre-programmed way. The values it sends are 1 minute apart also when it connects with Sibionics own app. For unknown reasons Sibionics own app transforms them into one in 5 minutes. Maybe it is because of the erratic appearance of the 1-minute curve.
• When you only look at the index%5==0 readings (60,65,70,75 ...), the glucose values in Juggluco are the same as those of Sibionics own app. If you have the Chinese Sibionics GS1 sensor, you can check that by taking a Chinese telephone number and using that app with the same sensor. It will also receive older values, which you can compare.
• You can get once in 5 minutes by using Sibionics own app. You will probably like it more.

**Reply by thejazzs** [reply]
Thank you for detailed information. 

I am using EU version of Sibionics. Getting every minute data would not a problem if it is not reducing the life of sensor.

Do you have any idea about the reading differences between blood readings as I mentioned what was going on detailed at my previous message. Actually this is important due to I am using calibration and cannot make a stable calibrate in case.

**Reply by j-kaltes** [reply]
You can't calibrate sensors in Sibionics own app. With what app did you calibrate? Didn't you use an app that was far off and had to correct that by calibration?

**Reply by thejazzs** [reply]
Sensor using with juggluco only (not using sibionics own app). 
Calibration with xdrip.
